packages/swayze/lib/src/widgets/headers/gestures/resize_header/resize_header_line.dart

```diff
@@ -0,0 +1,188 @@
+import 'package:cached_value/cached_value.dart';
+import 'package:flutter/widgets.dart';
+
+import '../../../../core/style/style.dart';
+
+/// Renders the resize header line.
+///
+/// See also:
+/// - [_ResizeHeaderLine].
+class ResizeHeaderLine extends StatelessWidget {
+  final SwayzeStyle style;
+  final Axis axis;
+
+  const ResizeHeaderLine({
+    Key? key,
+    required this.style,
+    required this.axis,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return _ResizeHeaderLine(
+      axis: axis,
+      fillColor: style.resizeHeaderStyle.fillColor,
+      lineColor: style.resizeHeaderStyle.lineColor,
+      thickness: style.cellSeparatorStrokeWidth,
+    );
+  }
+}
+
+/// A leaf render object that returns a [RenderBox] that paints an horizontal
+/// or vertical line (depending on the axis) with the given properties.
+///
+/// See also:
+/// - [_RenderResizeHeaderLine].
+class _ResizeHeaderLine extends LeafRenderObjectWidget {
```

Comment:
Since this is rendered in an overlay, couldn't it be a simple widget instead of having a custom render object?

Reply:
I didn't want to use negative values on the widget or on `Positioned`

Reply:
like a padding with `-5` for example on `left` property (same for `Positioned` widget)

Reply:
I see. What are the drawbacks of doing that?

Reply:
apart from guessing the appropriate padding for it (which I don't have to do in drawing in canvas, I literally just need the ancestor render box position), this could bring accessibility issues since it would require different paddings for different sizes of tables (this last thing is something that I can only speak for mobile, I'm not sure in desktop). I find easier to draw on canvas for this widget

Reply:
Thanks for the explanation 🙏